android/app/src/main/kotlin/coop/polypoly/polypod/updatenotification/UpdateNotification.kt

```diff
@@ -5,23 +5,65 @@ import coop.polypoly.polypod.Preferences
 import coop.polypoly.polypod.R
 
 class UpdateNotification(private val context: Context) {
-    val id = context.resources.getInteger(R.integer.update_notification_id)
+    companion object {
+        data class MockData(var id: Int? = null)
+        val mockData = MockData()
+    }
+
+    private enum class State {
+        NOT_SEEN,
+        PUSH_SEEN,
+        ALL_SEEN;
+
+        companion object {
+            fun parse(s: String?) =
+                s.let { name -> values().firstOrNull { it.name == name } }
+        }
+    }
+
+    val id = mockData.id
+        ?: context.resources.getInteger(R.integer.update_notification_id)
+
     val title = context.getString(R.string.update_notification_title)
+
     val text = context.getString(R.string.update_notification_text)
+
     val pushDelay =
         context.resources.getInteger(R.integer.update_notification_push_delay)
 
-    val inAppNotificationSeen: Boolean
-        get() = id == 0 || Preferences.getSeenInAppNotificationId(context) == id
+    val showPush: Boolean
+        get() = state == State.NOT_SEEN
+
+    val showInApp: Boolean
+        get() = state != State.ALL_SEEN
 
-    val pushNotificationSeen: Boolean
-        get() = id == 0 || Preferences.getSeenPushNotificationId(context) == id
+    private var state: State = loadLastState()
+        set(value) {
+            field = value
+            Preferences.setLastNotification(context, id, state.name)
+        }
+
+    private fun loadLastState(): State {
+        val (lastId, lastState) = Preferences.getLastNotification(context)
+        return when (id) {
+            0 -> State.ALL_SEEN
```

Comment:
Uh-oh, a specific ID value representing... stuff that is not an ID.

Reply:
Yeah, `0` is a special ID, which means: _Ignore this notification_.

Reply:
... also "reset the notification state so that you can start counting from 0 again"?

Reply:
No, notifications aren't really counted. What happens is this:

1. User installs polyPod, polyPod comes with a notification ID of, say, `1`. It's the very first time the polyPod is run, so the notification is marked as seen without actually being shown.
2. User installs new version of the polyPod, which comes with a notification ID of, say, `2`. It checks whether the previous notification ID is different (which it is, `1 !- 2`), and then shows the notification. Finally, it marks `2` as seen.

The notification ID is set by us, in `update_notification.xml`. Can be any valid integer - except `0`, which just means: _There is no notification_.